strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/test/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticationBase64EncodedTest.java

```diff
@@ -0,0 +1,134 @@
+
+    @Inject
+    private PasswordEncoder passwordEncoder;
+
+    @Test
+    public void base64EncodedPasswordAfterAlgorithmShouldWork()
+    {
+        UserDetails ldapUser = ldapUserDetailsService.loadUserByUsername("base64encoded-issue-1840");
+
+        assertThat(ldapUser).isInstanceOf(LdapUserDetailsImpl.class);
+
+        LdapUserDetails ldapUserDetails = (LdapUserDetailsImpl) ldapUser;
+
+        assertThat(ldapUserDetails.getDn()).isEqualTo("uid=base64encoded-issue-1840,ou=Users,dc=carlspring,dc=com");
+
+        assertThat(ldapUserDetails.getUsername()).isEqualTo("base64encoded-issue-1840");
+
+        assertThat(ldapUserDetails.getPassword()).isEqualTo("{MD5}5f4dcc3b5aa765d61d8327deb882cf99");
+
+        assertThat(passwordEncoder.matches("password", ldapUserDetails.getPassword())).isEqualTo(true);
+
+        assertThat(((List<SimpleGrantedAuthority>) ldapUser.getAuthorities()))
+                .contains(
+                        new SimpleGrantedAuthority(SystemRole.REPOSITORY_MANAGER.name()),
+                        new SimpleGrantedAuthority("USER_ROLE")
+                );
+
+    }
+
+    @Test
+    public void base64EncodedPasswordWithAlgorithmShouldWork()
+    {
+        UserDetails ldapUser = ldapUserDetailsService.loadUserByUsername("base64withalgorithm-issue-1840");
+
+        assertThat(ldapUser).isInstanceOf(LdapUserDetailsImpl.class);
+
+        LdapUserDetails ldapUserDetails = (LdapUserDetailsImpl) ldapUser;
+
+        assertThat(ldapUserDetails.getDn()).isEqualTo(
+                "uid=base64withalgorithm-issue-1840,ou=Users,dc=carlspring,dc=com");
+
+        assertThat(ldapUserDetails.getUsername()).isEqualTo("base64withalgorithm-issue-1840");
+
+        assertThat(ldapUserDetails.getPassword()).isEqualTo(
+                "{bcrypt}$2a$10$lpwlxyjvXKzN1ccCrw2PBuZx.eVesWbfmTbsrCboMU.gsNWVcZWMi");
+
+        assertThat(passwordEncoder.matches("password", ldapUserDetails.getPassword())).isEqualTo(true);
+
+        assertThat(((List<SimpleGrantedAuthority>) ldapUser.getAuthorities()))
+                .contains(
+                        new SimpleGrantedAuthority(SystemRole.REPOSITORY_MANAGER.name()),
+                        new SimpleGrantedAuthority("USER_ROLE")
+                );
+
+    }
+
+    public static class TestContextInitializer
+            extends AuthenticationContextInitializer
+    {
+
+        public TestContextInitializer()
+        {
+            super(loadPropertySource());
+        }
+
+        private static PropertySource<?> loadPropertySource()
+        {
+            Properties properties = new Properties();
+            try (InputStream is = LdapAuthenticationProviderTest.class.getResourceAsStream("lapt.properties"))
+            {
+                properties.load(is);
+            }
+            catch (Exception ex)
+            {
+                throw new UndeclaredThrowableException(ex);
+            }
+
+            properties.put(BASE64_ENCODING_KEY, true);
```

Comment:
@madhukarbharti you shouldn't do it like this here as well. Please use another properties file for this test instead. It should be `labet.properties` and have `strongbox.authentication.ldap.base64EncodedPassword=true`.